internal/server/router/meter.go

```diff
@@ -3,23 +3,27 @@ package router
 import (
 	"errors"
 	"fmt"
-	"log/slog"
 	"net/http"
 
 	"github.com/go-chi/render"
 
+	"github.com/openmeterio/openmeter/pkg/contextx"
 	"github.com/openmeterio/openmeter/pkg/models"
 	"github.com/openmeterio/openmeter/pkg/slicesx"
 )
 
 func (a *Router) ListMeters(w http.ResponseWriter, r *http.Request) {
-	logger := slog.With("operation", "listMeters")
+	ctx := contextx.WithAttr(r.Context(), "operation", "listMeters")
+
 	namespace := a.config.NamespaceManager.GetDefaultNamespace()
 
 	meters, err := a.config.Meters.ListMeters(r.Context(), namespace)
 	if err != nil {
 		err := fmt.Errorf("list meters: %w", err)
-		models.NewStatusProblem(r.Context(), err, http.StatusInternalServerError).Respond(logger, w, r)
+
+		a.config.ErrorHandler.HandleContext(ctx, err)
```

Comment:
this is such a commonly used function, wondering if we could add it to router and just be `a.HandleError(ctx, err))` or bake it it into NewStatusProblem.Respond

Reply:
If we add anything back to the status problem, I'm gonna have to cry myself to sleep. 😄 

The right solution is not baking business logic into HTTP handlers. Then we don't have to handle and map every error manually every single time, but we can rely on middleware instead.

Adding logging to the status problem was always a half measure (if you can call it that). If it doesn't hurt enough though, we are never going to fix it.

tl;dr The explicitness will lead us to a better solution eventually.